### PR TITLE
web, website: compress images

### PR DESCRIPTION
**Pull request by rissson**
## Details

Compress images. This should be lossless. Size gain: 88M -> 16M.

Somewhat part of https://github.com/goauthentik/meta/issues/20, as I'll also be excluding the `.git` folder from the docker build. 

### New Features

-   Adds feature which does x, y, and z.

### Breaking Changes

-   Adds breaking change which causes \<issue\>.

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)
-   [ ] The translation files have been updated (`make i18n-extract`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

### CODE REVIEW

**Comment by rissson** [comment]
Do we want to automatically push those changes on the PR branch?

File: .github/workflows/compress-images.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: compress-images
+
+on:
+  pull_request:
+    branches:
+      - main
```
Comment (rissson) [inline .github/workflows/compress-images.yml:6]:
```suggestion
      - main
    paths:
      - "**/*.jpg"
      - "**/*.png"
```

File: .github/workflows/compress-images.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  compress:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          sudo apt install -y imagemagick optipng 
+          find -name '*.jpg' -exec mogrify -strip -compress JPEG {} +
```
Comment (BeryJu) [inline .github/workflows/compress-images.yml:15]: we already use https://sharp.pixelplumbing.com/ for the flow background (see https://github.com/goauthentik/authentik/blob/main/web/package.json#L25) so maybe we should use that here too?

Also I dont think these compression commands are idempotent so this CI pipeline might always fail?

**Review by BeryJu — COMMENTED** [review]
I don't think we'll get much of a size improvement from the image compression (at least from the potential pain of making a CI pipeline for that), I'm curious of how much of a difference ignoring .git makes with docker

**Comment by rissson** [comment]
> I don't think we'll get much of a size improvement from the image compression

The size reduction by compressing images is 72M, which makes our docker image go from 832M to 760M, i.e. a 9% improvement.

> at least from the potential pain of making a CI pipeline for that

We can also not make a CI pipeline and just have this as a one-off.

> I'm curious of how much of a difference ignoring .git makes with docker

Nothing in terms of image size now that I looked at it more closely, as we don't do `COPY ./ ./`, it would just be an improvement of not copying the `.git` directory over to the docker daemon when building.

File: .github/workflows/compress-images.yml (hunk 2 above)
Comment (rissson) [reply .github/workflows/compress-images.yml:15]: optipng is idempotent, imagemagick (either `convert` or `mogrify`) isn't

**Comment by rissson** [comment]
Implement https://github.com/marketplace/actions/image-actions

**Comment by rissson** [comment]
Unfortunately, I cannot customize the commit name :/

**Review by BeryJu — APPROVED** [review]
LGTM, commit message doesn't really matter since we squash merge anyways

I copied all the current state icons/etc to https://github.com/goauthentik/brand-assets to keep the original copies (not that it matters too much since we have svgs but anyways)

File: .github/workflows/image-compress.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        id: cpr
+        with:
+          token: ${{ steps.generate_token.outputs.token }}
+          title: "*: Auto compress images"
+          branch-suffix: timestamp
+          commit-messsage: "*: compress images"
+          body: ${{ steps.compress.outputs.markdown }}
+          delete-branch: true
+          signoff: true
+      - uses: peter-evans/enable-pull-request-automerge@v3
```
[the remark this answers is not in the document]
Comment (BeryJu) [reply .github/workflows/image-compress.yml:56]: 👍 I was just about to push that exact same change too lol

[the remark this answers is not in the document]
Comment (rissson) [reply .github/workflows/image-compress.yml:56]: Yes, for once I was faster :tada:

**Pull request merged**